### PR TITLE
[PA-227] Added retry count and retry cooldown settings for Jira calls

### PR DESCRIPTION
**Pull request by umutozk**
#### Changes
- Added retry mechanism to jira calls.

### CODE REVIEW

**Comment by emil-cazacu-aurea** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

File: src/main/java/com/devfactory/processautomation/service/process/SheetProcessingServiceImpl.java [hunk 1]
```diff
@@ -83,14 +89,28 @@ public ProcessResult process(final ProcessDto processDto) {
...
-                            logger.error("Failed to create ticket for row " + ticket.getSheetRowNumber());
-                            ticket.setResultText(getErrorsString(response.getErrors()));
-                            errors.addAll(response.getErrors());
-                        } else {
-                            logger.info("Ticket created for row " + ticket.getSheetRowNumber());
-                            ticket.setResultText(response.getTicketKey());
+                        for (int i = 0; i < appConfig.getRetryCount(); i++) {
+                            final TicketResponse response = jiraClient.createIssue(ticket.getTicket());
+                            if (!response.isValid()) {
+                                logger.error(String
```
Comment (josemartinsquest) [inline src/main/java/com/devfactory/processautomation/service/process/SheetProcessingServiceImpl.java]: Decouple to HandleFailure() here.

The cool down it self should be inside.

File: src/main/java/com/devfactory/processautomation/service/process/SheetProcessingServiceImpl.java [hunk 2]
```diff
@@ -83,14 +89,28 @@ public ProcessResult process(final ProcessDto processDto) {
...
+                                    .format("Failed to create ticket for row %d | Try %d", ticket.getSheetRowNumber(),
+                                        i + 1));
+                                if (i == appConfig.getRetryCount() - 1) {
+                                    ticket.setResultText(getErrorsString(response.getErrors()));
+                                    errors.addAll(response.getErrors());
+                                }
+                            } else {
+                                logger.info("Ticket created for row " + ticket.getSheetRowNumber());
```
Comment (josemartinsquest) [inline src/main/java/com/devfactory/processautomation/service/process/SheetProcessingServiceImpl.java]: HandleSuccess()

**Comment by emil-cazacu-aurea** [comment]
:white_check_mark: No issues identified in this pull request. 
    
**Good work!**

**Comment by emil-cazacu-aurea** [comment]
Hi, I'm Emil Cazacu from [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members) and I have just reviewed your PR.
<b>How likely would you recommend the Code Review service to another developer?</b>

Not at all likely \[ [1](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/4&entry.1163746017=1) \] \[ [2](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/4&entry.1163746017=2) \] \[ [3](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/4&entry.1163746017=3) \] \[ [4](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/4&entry.1163746017=4) \] \[ [5](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/4&entry.1163746017=5) \] \[ [6](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/4&entry.1163746017=6) \] \[ [7](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/4&entry.1163746017=7) \] \[ [8](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/4&entry.1163746017=8) \] \[ [9](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/4&entry.1163746017=9) \] \[ [10](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/DF-PA-Projects-RC-SheetToJiraTool/pull/4&entry.1163746017=10) \] Very likely

> TIP: Press \`Ctrl\` (or \`⌘\` in a Mac) to open the survey in a new tab.

File: src/main/java/com/devfactory/processautomation/service/process/SheetProcessingServiceImpl.java [hunk 3]
```diff
@@ -109,6 +115,31 @@ public ProcessResult process(final ProcessDto processDto) {
...
+    private List<ValidationError> handleFailure(final int iteration, final TicketDto ticket,
+        final List<ValidationError> responseErrors) {
+        logger.error(String
+            .format("Failed to create ticket for row %d | Try %d", ticket.getSheetRowNumber(),
+                iteration + 1));
+        if (iteration == appConfig.getRetryCount() - 1) {
+            ticket.setResultText(getErrorsString(responseErrors));
+            return responseErrors;
+        } else {
+            try {
```
Comment (josemartinsquest) [inline src/main/java/com/devfactory/processautomation/service/process/SheetProcessingServiceImpl.java]: Decouple this to CollDown();

**Pull request merged**